pkg/util/test/util.go

```diff
@@ -2191,6 +2189,26 @@ func validateCSISnapshotController(cluster *corev1.StorageCluster, pxImageList m
 	deployment.Namespace = cluster.Namespace
 	deployment.Name = "px-csi-ext"
 	t := func() (interface{}, bool, error) {
+		// Check whether snapshot controller container should be installed
+		installSnapshotController := true
+		podList, err := coreops.Instance().ListPods(nil)
+		if err != nil {
+			return nil, true, fmt.Errorf("failed to list pods from all namespaces")
+		}
+		for _, p := range podList.Items {
+			// ignore pods deployed by operator
+			if label, ok := p.Labels["app"]; ok && label == "px-csi-driver" {
+				continue
+			}
+			for _, c := range p.Spec.Containers {
+				if strings.Contains(c.Image, "/snapshot-controller:") {
+					logrus.Infof("found external snapshot controller in pod %s/%s", p.Namespace, p.Name)
+					installSnapshotController = false
```

Comment:
we should verify if external snapshot controller exists, operator installed snapshot controller should not exist.

Reply:
yes, the validation is already underneath, line 2228, it would fail if found the snapshot controller deployed in csi deployment.